### PR TITLE
Generate lambdas instead of anonymous functions with Java 8 flag

### PR DESCRIPTION
**Pull request by daschult**

### CODE REVIEW

**Comment by olydis** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.3.14*

## ❌️ General: 92%

<details><summary>38 out of 522 features are not covered by tests</summary><p>

[`ConstantsInBody`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInBody), [`ConstantsInPath`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInPath), [`FileStreamVeryLarge`](https://github.com/Azure/autorest.testserver/search?q=FileStreamVeryLarge), [`FormdataStreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=FormdataStreamUploadFile), [`HttpRedirect300Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Get), [`HttpRedirect300Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Head), [`HttpRedirect301Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Get), [`HttpRedirect301Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Put), [`HttpRedirect302Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Get), [`HttpRedirect302Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Head), [`HttpRedirect302Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Patch), [`HttpRedirect303Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect303Post), [`HttpRedirect307Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Delete), [`HttpRedirect307Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Get), [`HttpRedirect307Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Head), [`HttpRedirect307Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Patch), [`HttpRedirect307Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Post), [`HttpRedirect307Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Put), [`ResponsesScenarioC200MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC200MatchingModel), [`ResponsesScenarioC201MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC201MatchingModel), [`ResponsesScenarioC404MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC404MatchingModel), [`StreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=StreamUploadFile), [`UrlPathsArrayCSVInPath`](https://github.com/Azure/autorest.testserver/search?q=UrlPathsArrayCSVInPath), [`UrlQueriesArrayMultiEmpty`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiEmpty), [`UrlQueriesArrayMultiNull`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiNull), [`UrlQueriesArrayMultiValid`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiValid), [`allowedValueEnum`](https://github.com/Azure/autorest.testserver/search?q=allowedValueEnum), [`expectedEnum`](https://github.com/Azure/autorest.testserver/search?q=expectedEnum), [`getByteEmpty`](https://github.com/Azure/autorest.testserver/search?q=getByteEmpty), [`getByteInvalid`](https://github.com/Azure/autorest.testserver/search?q=getByteInvalid), [`getComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=getComplexPolymorphismComplicated), [`getEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferenced), [`getEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferencedConstant), [`putComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismComplicated), [`putEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferenced), [`putEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferencedConstant), [`roundTripEnum`](https://github.com/Azure/autorest.testserver/search?q=roundTripEnum), [`unexpectedEnum`](https://github.com/Azure/autorest.testserver/search?q=unexpectedEnum)</p></details>

## ❌️ Azure: 94%

<details><summary>7 out of 118 features are not covered by tests</summary><p>

[`AzureMethodQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureMethodQueryUrlEncoding), [`AzureODataFilter`](https://github.com/Azure/autorest.testserver/search?q=AzureODataFilter), [`AzurePathQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzurePathQueryUrlEncoding), [`AzureRequestClientIdInError`](https://github.com/Azure/autorest.testserver/search?q=AzureRequestClientIdInError), [`AzureSwaggerQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureSwaggerQueryUrlEncoding), [`AzureXmsRequestClientIdNull`](https://github.com/Azure/autorest.testserver/search?q=AzureXmsRequestClientIdNull), [`LRONonRetryPut201Creating400InvalidJson`](https://github.com/Azure/autorest.testserver/search?q=LRONonRetryPut201Creating400InvalidJson)</p></details>

**Comment by daschult** [comment]
> linux-x64 restart

**Comment by daschult** [comment]
> win32-x64 restart

**Comment by olydis** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.3.14*

## ❌️ General: 92%

<details><summary>38 out of 522 features are not covered by tests</summary><p>

[`ConstantsInBody`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInBody), [`ConstantsInPath`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInPath), [`FileStreamVeryLarge`](https://github.com/Azure/autorest.testserver/search?q=FileStreamVeryLarge), [`FormdataStreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=FormdataStreamUploadFile), [`HttpRedirect300Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Get), [`HttpRedirect300Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Head), [`HttpRedirect301Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Get), [`HttpRedirect301Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Put), [`HttpRedirect302Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Get), [`HttpRedirect302Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Head), [`HttpRedirect302Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Patch), [`HttpRedirect303Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect303Post), [`HttpRedirect307Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Delete), [`HttpRedirect307Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Get), [`HttpRedirect307Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Head), [`HttpRedirect307Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Patch), [`HttpRedirect307Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Post), [`HttpRedirect307Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Put), [`ResponsesScenarioC200MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC200MatchingModel), [`ResponsesScenarioC201MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC201MatchingModel), [`ResponsesScenarioC404MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC404MatchingModel), [`StreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=StreamUploadFile), [`UrlPathsArrayCSVInPath`](https://github.com/Azure/autorest.testserver/search?q=UrlPathsArrayCSVInPath), [`UrlQueriesArrayMultiEmpty`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiEmpty), [`UrlQueriesArrayMultiNull`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiNull), [`UrlQueriesArrayMultiValid`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiValid), [`allowedValueEnum`](https://github.com/Azure/autorest.testserver/search?q=allowedValueEnum), [`expectedEnum`](https://github.com/Azure/autorest.testserver/search?q=expectedEnum), [`getByteEmpty`](https://github.com/Azure/autorest.testserver/search?q=getByteEmpty), [`getByteInvalid`](https://github.com/Azure/autorest.testserver/search?q=getByteInvalid), [`getComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=getComplexPolymorphismComplicated), [`getEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferenced), [`getEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferencedConstant), [`putComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismComplicated), [`putEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferenced), [`putEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferencedConstant), [`roundTripEnum`](https://github.com/Azure/autorest.testserver/search?q=roundTripEnum), [`unexpectedEnum`](https://github.com/Azure/autorest.testserver/search?q=unexpectedEnum)</p></details>

## ❌️ Azure: 94%

<details><summary>7 out of 118 features are not covered by tests</summary><p>

[`AzureMethodQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureMethodQueryUrlEncoding), [`AzureODataFilter`](https://github.com/Azure/autorest.testserver/search?q=AzureODataFilter), [`AzurePathQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzurePathQueryUrlEncoding), [`AzureRequestClientIdInError`](https://github.com/Azure/autorest.testserver/search?q=AzureRequestClientIdInError), [`AzureSwaggerQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureSwaggerQueryUrlEncoding), [`AzureXmsRequestClientIdNull`](https://github.com/Azure/autorest.testserver/search?q=AzureXmsRequestClientIdNull), [`LRONonRetryPut201Creating400InvalidJson`](https://github.com/Azure/autorest.testserver/search?q=LRONonRetryPut201Creating400InvalidJson)</p></details>

**Comment by olydis** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.3.14*

## ❌️ General: 0%

no tests were run for this category


## ❌️ Azure: 94%

<details><summary>7 out of 118 features are not covered by tests</summary><p>

[`AzureMethodQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureMethodQueryUrlEncoding), [`AzureODataFilter`](https://github.com/Azure/autorest.testserver/search?q=AzureODataFilter), [`AzurePathQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzurePathQueryUrlEncoding), [`AzureRequestClientIdInError`](https://github.com/Azure/autorest.testserver/search?q=AzureRequestClientIdInError), [`AzureSwaggerQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureSwaggerQueryUrlEncoding), [`AzureXmsRequestClientIdNull`](https://github.com/Azure/autorest.testserver/search?q=AzureXmsRequestClientIdNull), [`LRONonRetryPut201Creating400InvalidJson`](https://github.com/Azure/autorest.testserver/search?q=LRONonRetryPut201Creating400InvalidJson)</p></details>

**Comment by olydis** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.3.14*

## ❌️ General: 0%

no tests were run for this category


## ❌️ Azure: 94%

<details><summary>7 out of 118 features are not covered by tests</summary><p>

[`AzureMethodQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureMethodQueryUrlEncoding), [`AzureODataFilter`](https://github.com/Azure/autorest.testserver/search?q=AzureODataFilter), [`AzurePathQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzurePathQueryUrlEncoding), [`AzureRequestClientIdInError`](https://github.com/Azure/autorest.testserver/search?q=AzureRequestClientIdInError), [`AzureSwaggerQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureSwaggerQueryUrlEncoding), [`AzureXmsRequestClientIdNull`](https://github.com/Azure/autorest.testserver/search?q=AzureXmsRequestClientIdNull), [`LRONonRetryPut201Creating400InvalidJson`](https://github.com/Azure/autorest.testserver/search?q=LRONonRetryPut201Creating400InvalidJson)</p></details>

**Comment by RikkiGibson** [comment]
Seeing CI compile errors such as
```autorest.java/test/javaversion-7/src/test/java/fixtures/paging/PagingTests.java:[6,44] package fixtures.javaversion7.implementation does not exist```

**Comment by RikkiGibson** [comment]
For something like a Java 8 versus Java 7 flag to provide value we would need to start shipping separate Java 8 and Java 7 versions of the runtime. Let's drop the idea of a flag for this and simply start converting to use a Java 8 requirement.

**Comment by daschult** [comment]
@RikkiGibson I removed the `javaversion` flag and we're now defaulting to Java 8.
I also extracted lambda creation out into the JavaBlock class. I've found that we tend to have trouble with our generator creating incorrect Java code when we handwrite too much of the Java code and don't let the Java generator framework (such as JavaBlock) take care of creating the Java code.
I added the lambda parameter types back in because I had no idea what type `res` was in all of the places it was defined. I generally don't like type-less lambda headers, so unless you can come up with a compelling reason to no having them, I'd prefer to keep them in all of the generated code.

**Comment by olydis** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.3.14*

## ❌️ General: 92%

<details><summary>38 out of 522 features are not covered by tests</summary><p>

[`ConstantsInBody`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInBody), [`ConstantsInPath`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInPath), [`FileStreamVeryLarge`](https://github.com/Azure/autorest.testserver/search?q=FileStreamVeryLarge), [`FormdataStreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=FormdataStreamUploadFile), [`HttpRedirect300Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Get), [`HttpRedirect300Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Head), [`HttpRedirect301Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Get), [`HttpRedirect301Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Put), [`HttpRedirect302Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Get), [`HttpRedirect302Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Head), [`HttpRedirect302Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Patch), [`HttpRedirect303Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect303Post), [`HttpRedirect307Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Delete), [`HttpRedirect307Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Get), [`HttpRedirect307Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Head), [`HttpRedirect307Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Patch), [`HttpRedirect307Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Post), [`HttpRedirect307Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Put), [`ResponsesScenarioC200MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC200MatchingModel), [`ResponsesScenarioC201MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC201MatchingModel), [`ResponsesScenarioC404MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC404MatchingModel), [`StreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=StreamUploadFile), [`UrlPathsArrayCSVInPath`](https://github.com/Azure/autorest.testserver/search?q=UrlPathsArrayCSVInPath), [`UrlQueriesArrayMultiEmpty`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiEmpty), [`UrlQueriesArrayMultiNull`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiNull), [`UrlQueriesArrayMultiValid`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiValid), [`allowedValueEnum`](https://github.com/Azure/autorest.testserver/search?q=allowedValueEnum), [`expectedEnum`](https://github.com/Azure/autorest.testserver/search?q=expectedEnum), [`getByteEmpty`](https://github.com/Azure/autorest.testserver/search?q=getByteEmpty), [`getByteInvalid`](https://github.com/Azure/autorest.testserver/search?q=getByteInvalid), [`getComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=getComplexPolymorphismComplicated), [`getEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferenced), [`getEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferencedConstant), [`putComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismComplicated), [`putEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferenced), [`putEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferencedConstant), [`roundTripEnum`](https://github.com/Azure/autorest.testserver/search?q=roundTripEnum), [`unexpectedEnum`](https://github.com/Azure/autorest.testserver/search?q=unexpectedEnum)</p></details>

## ❌️ Azure: 94%

<details><summary>7 out of 118 features are not covered by tests</summary><p>

[`AzureMethodQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureMethodQueryUrlEncoding), [`AzureODataFilter`](https://github.com/Azure/autorest.testserver/search?q=AzureODataFilter), [`AzurePathQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzurePathQueryUrlEncoding), [`AzureRequestClientIdInError`](https://github.com/Azure/autorest.testserver/search?q=AzureRequestClientIdInError), [`AzureSwaggerQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureSwaggerQueryUrlEncoding), [`AzureXmsRequestClientIdNull`](https://github.com/Azure/autorest.testserver/search?q=AzureXmsRequestClientIdNull), [`LRONonRetryPut201Creating400InvalidJson`](https://github.com/Azure/autorest.testserver/search?q=LRONonRetryPut201Creating400InvalidJson)</p></details>

**Comment by olydis** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.3.14*

## ❌️ General: 92%

<details><summary>38 out of 522 features are not covered by tests</summary><p>

[`ConstantsInBody`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInBody), [`ConstantsInPath`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInPath), [`FileStreamVeryLarge`](https://github.com/Azure/autorest.testserver/search?q=FileStreamVeryLarge), [`FormdataStreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=FormdataStreamUploadFile), [`HttpRedirect300Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Get), [`HttpRedirect300Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Head), [`HttpRedirect301Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Get), [`HttpRedirect301Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Put), [`HttpRedirect302Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Get), [`HttpRedirect302Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Head), [`HttpRedirect302Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Patch), [`HttpRedirect303Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect303Post), [`HttpRedirect307Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Delete), [`HttpRedirect307Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Get), [`HttpRedirect307Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Head), [`HttpRedirect307Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Patch), [`HttpRedirect307Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Post), [`HttpRedirect307Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Put), [`ResponsesScenarioC200MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC200MatchingModel), [`ResponsesScenarioC201MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC201MatchingModel), [`ResponsesScenarioC404MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC404MatchingModel), [`StreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=StreamUploadFile), [`UrlPathsArrayCSVInPath`](https://github.com/Azure/autorest.testserver/search?q=UrlPathsArrayCSVInPath), [`UrlQueriesArrayMultiEmpty`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiEmpty), [`UrlQueriesArrayMultiNull`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiNull), [`UrlQueriesArrayMultiValid`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiValid), [`allowedValueEnum`](https://github.com/Azure/autorest.testserver/search?q=allowedValueEnum), [`expectedEnum`](https://github.com/Azure/autorest.testserver/search?q=expectedEnum), [`getByteEmpty`](https://github.com/Azure/autorest.testserver/search?q=getByteEmpty), [`getByteInvalid`](https://github.com/Azure/autorest.testserver/search?q=getByteInvalid), [`getComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=getComplexPolymorphismComplicated), [`getEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferenced), [`getEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferencedConstant), [`putComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismComplicated), [`putEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferenced), [`putEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferencedConstant), [`roundTripEnum`](https://github.com/Azure/autorest.testserver/search?q=roundTripEnum), [`unexpectedEnum`](https://github.com/Azure/autorest.testserver/search?q=unexpectedEnum)</p></details>

## ❌️ Azure: 94%

<details><summary>7 out of 118 features are not covered by tests</summary><p>

[`AzureMethodQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureMethodQueryUrlEncoding), [`AzureODataFilter`](https://github.com/Azure/autorest.testserver/search?q=AzureODataFilter), [`AzurePathQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzurePathQueryUrlEncoding), [`AzureRequestClientIdInError`](https://github.com/Azure/autorest.testserver/search?q=AzureRequestClientIdInError), [`AzureSwaggerQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureSwaggerQueryUrlEncoding), [`AzureXmsRequestClientIdNull`](https://github.com/Azure/autorest.testserver/search?q=AzureXmsRequestClientIdNull), [`LRONonRetryPut201Creating400InvalidJson`](https://github.com/Azure/autorest.testserver/search?q=LRONonRetryPut201Creating400InvalidJson)</p></details>

File: regenerate.cmd [hunk 1]
```diff
@@ -0,0 +1,66 @@
+@echo off
```
Comment (RikkiGibson) [inline regenerate.cmd]: Is `gulp regenerate` still having problems?

**Comment by RikkiGibson** [comment]
Only reason I'd argue for eliding them in gen code is if it were somehow a hassle to get the type name or add the appropriate extra imports in the context of emitting the lambda.

**Comment by olydis** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.3.14*

## ❌️ General: 92%

<details><summary>38 out of 522 features are not covered by tests</summary><p>

[`ConstantsInBody`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInBody), [`ConstantsInPath`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInPath), [`FileStreamVeryLarge`](https://github.com/Azure/autorest.testserver/search?q=FileStreamVeryLarge), [`FormdataStreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=FormdataStreamUploadFile), [`HttpRedirect300Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Get), [`HttpRedirect300Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Head), [`HttpRedirect301Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Get), [`HttpRedirect301Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Put), [`HttpRedirect302Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Get), [`HttpRedirect302Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Head), [`HttpRedirect302Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Patch), [`HttpRedirect303Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect303Post), [`HttpRedirect307Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Delete), [`HttpRedirect307Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Get), [`HttpRedirect307Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Head), [`HttpRedirect307Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Patch), [`HttpRedirect307Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Post), [`HttpRedirect307Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Put), [`ResponsesScenarioC200MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC200MatchingModel), [`ResponsesScenarioC201MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC201MatchingModel), [`ResponsesScenarioC404MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC404MatchingModel), [`StreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=StreamUploadFile), [`UrlPathsArrayCSVInPath`](https://github.com/Azure/autorest.testserver/search?q=UrlPathsArrayCSVInPath), [`UrlQueriesArrayMultiEmpty`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiEmpty), [`UrlQueriesArrayMultiNull`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiNull), [`UrlQueriesArrayMultiValid`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiValid), [`allowedValueEnum`](https://github.com/Azure/autorest.testserver/search?q=allowedValueEnum), [`expectedEnum`](https://github.com/Azure/autorest.testserver/search?q=expectedEnum), [`getByteEmpty`](https://github.com/Azure/autorest.testserver/search?q=getByteEmpty), [`getByteInvalid`](https://github.com/Azure/autorest.testserver/search?q=getByteInvalid), [`getComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=getComplexPolymorphismComplicated), [`getEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferenced), [`getEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferencedConstant), [`putComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismComplicated), [`putEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferenced), [`putEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferencedConstant), [`roundTripEnum`](https://github.com/Azure/autorest.testserver/search?q=roundTripEnum), [`unexpectedEnum`](https://github.com/Azure/autorest.testserver/search?q=unexpectedEnum)</p></details>

## ❌️ Azure: 94%

<details><summary>7 out of 118 features are not covered by tests</summary><p>

[`AzureMethodQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureMethodQueryUrlEncoding), [`AzureODataFilter`](https://github.com/Azure/autorest.testserver/search?q=AzureODataFilter), [`AzurePathQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzurePathQueryUrlEncoding), [`AzureRequestClientIdInError`](https://github.com/Azure/autorest.testserver/search?q=AzureRequestClientIdInError), [`AzureSwaggerQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureSwaggerQueryUrlEncoding), [`AzureXmsRequestClientIdNull`](https://github.com/Azure/autorest.testserver/search?q=AzureXmsRequestClientIdNull), [`LRONonRetryPut201Creating400InvalidJson`](https://github.com/Azure/autorest.testserver/search?q=LRONonRetryPut201Creating400InvalidJson)</p></details>

File: regenerate.cmd (hunk 1 above)
Comment (daschult) [reply regenerate.cmd]: It is on my desktop. I'm going to install the latest version of node.js and see if that helps.

Comment (daschult) [reply regenerate.cmd]: Even with the latest node.js I'm still having issues. I'll merge this pull request and then start working on figuring out what is going on with my machine.

**Pull request merged**